### ISSUE TITLE
Unable to switch MPI library on Pkg.build()

### ISSUE THREAD

**Issue by ptooley** [JuliaParallel/MPI.jl#373]
Documented behaviour is that rerunning `Pkg.build("MPI")` with `JULIA_MPI_LIBRARY` set should allow changing of the MPI library that is used at runtime, however that is not currently functioning:

```shell
$ julia -E 'using MPI; MPI.libmpi'
  "/opt/spack/linux-gentoo2-zen2/gcc-9.3.0/extrae-3.7.1-756jeol5vjm6nya56ua2l7g4oxw77hzc/lib/libmpitracecf.so"
$ export JULIA_MPI_LIBRARY="/usr/lib64/libmpi.so"
$ julia -e 'using Pkg; Pkg.build("MPI")'
$ julia -E 'using MPI; MPI.libmpi'
  "/opt/spack/linux-gentoo2-zen2/gcc-9.3.0/extrae-3.7.1-756jeol5vjm6nya56ua2l7g4oxw77hzc/lib/libmpitracecf.so"  <-- I expected this to have changed...
```

Julia Version: 1.4.0
MPI Version: 0.13.1

**Comment by simonbyrne** [comment]
Fixed by #376.

**Comment by ptooley** [comment]
Thank you, much appreciated!
